include/swift-c/DependencyScan/DependencyScan.h

```diff
@@ -445,22 +448,46 @@ typedef enum {
   SWIFTSCAN_OUTPUT_TYPE_CLANG_PCH = 5
 } swiftscan_output_kind_t;
 
-/// Create a \c cas instance that points to path.
-SWIFTSCAN_PUBLIC swiftscan_cas_t swiftscan_cas_create(const char *path);
+/// Create a \c CASOptions for creating CAS inside scanner specified.
+SWIFTSCAN_PUBLIC swiftscan_cas_options_t swiftscan_cas_options_create(void);
+
+/// Dispose \c CASOptions.
+SWIFTSCAN_PUBLIC void
+swiftscan_cas_options_dispose(swiftscan_cas_options_t options);
+
+/// Set on-disk path for the \c cas.
+SWIFTSCAN_PUBLIC void
+swiftscan_cas_options_set_ondisk_path(swiftscan_cas_options_t options,
+                                      const char *path);
+
+/// Set plugin path for the \c cas.
+SWIFTSCAN_PUBLIC void
+swiftscan_cas_options_set_plugin_path(swiftscan_cas_options_t options,
+                                      const char *path);
+
+/// Set option using a name/value pair. Return true if error.
+SWIFTSCAN_PUBLIC bool
+swiftscan_cas_options_set_option(swiftscan_cas_options_t options,
+                                 const char *name, const char *value,
+                                 swiftscan_string_ref_t *error);
+
+/// Create a \c cas instance from plugin.
+SWIFTSCAN_PUBLIC swiftscan_cas_t swiftscan_cas_create_from_options(
+    swiftscan_cas_options_t options, swiftscan_string_ref_t *error);
 
 /// Dispose the \c cas instance.
 SWIFTSCAN_PUBLIC void swiftscan_cas_dispose(swiftscan_cas_t cas);
 
 /// Store content into CAS. Return \c CASID as string.
-SWIFTSCAN_PUBLIC swiftscan_string_ref_t swiftscan_cas_store(swiftscan_cas_t cas,
-                                                            uint8_t *data,
-                                                            unsigned size);
+SWIFTSCAN_PUBLIC swiftscan_string_ref_t
```

Comment:
Could you clarify in the documentation comments that the caller is supposed to dispose the strings returned by the APIs by calling `swiftscan_string_dispose`?
The documentation of `swiftscan_string_ref_t` implies that its lifetime is tied to another object which can be confusing.